packages/node-runtime-worker-thread/src/spawn-child-from-source.ts

```diff
@@ -25,6 +25,7 @@ export default function spawnChildFromSource(
   _stdout: StdioNull | StdioPipe = 'inherit',
   _stderr: StdioNull | StdioPipe = 'inherit'
 ): Promise<ChildProcess> {
+  // eslint-disable-next-line no-async-promise-executor
```

Comment:
I think there is only one place in this function where await is used, maybe we can just change it to

```ts
if (!childProcess.stdin) {
  void kill(childProcess).then(() => {
    reject(
      new Error("Can't write src to the spawned process, missing stdin")
    );
  });
  return;
}
```

or something similar?

Reply:
I made that change but then I also have to add a catch or ignore more things.. Which makes me think: What would happen if that await threw an error? I think it was unhandled already so I can just catch and throw to get back the same behaviour, right?

Reply:
Hmmm, wasn't sure what exactly will happen, tried it out, and yeah, seems like it will cause an unhandled rejection and the promise returned from `spawnChildFromSource` will never resolve, yikes! Yeah, I think rejecting when error is encountered in kill will be the best thing to do here

Reply:
Changed it to reject.